docs/documentation/api.md

```diff
@@ -254,13 +254,13 @@ above.
    question works by reading the code! To understand how arguments
    are specified in Zulip backend endpoints, read our [REST API
    tutorial][rest-api-tutorial], paying special attention to the
-   details of `REQ` and `has_request_variables`.
+   details of `Annotated` and `typed_endpoint`.
```

Comment:
The most similar component from the `typed_endpoint` infrastructure to `REQ` is probably `ApiParamConfig`. Here I think mentioning `typed_endpoint` should be sufficient.